### ISSUE TITLE
Qt5Agg eats 100% CPU when plotting with block=True in interactive mode

### ISSUE THREAD

**Issue by Rufflewind** [matplotlib/matplotlib#10140]
### Bug report

**Bug summary**

When using Qt5Agg, Matplotlib consumes a full CPU core while an interactive window is open.  The cursor repeatedly shows a "spinner" when I mouse over the canvas.  This didn't used to be the case.

**Code for reproduction**

```python
import matplotlib.pyplot as plt
plt.plot([1, 2], [3, 4])
plt.ion()
plt.show(block=True)
```

**Actual outcome**

One of the CPU cores is pegged at 100%.  Interface is very slow and barely responsive.

**Expected outcome**

Should not use up 100% of a CPU core when nothing is happening.

**Matplotlib version**

  * Operating system: Linux 4.14.8
  * Matplotlib version: 2.1.1+864.g92a93d698.dirty
  * Matplotlib backend: Qt5Agg
  * Python version: 3.6.4

This was reproduced on both XMonad and XFCE4.  Affects both `python-matplotlib` (2.1.1) from official Arch repos and `python-matplotlib-git` from AUR.

**Comment by tacaswell** [comment]
This is caused by https://github.com/matplotlib/matplotlib/pull/9505 which fixed https://github.com/matplotlib/matplotlib/issues/9406 .

@Rufflewind Thanks for the clear report on this.  We have seen this tangentially in a couple of other bug reports, but this made it very clear what was going wrong!

**Comment by dstansby** [comment]
Fixed by #10142
